### PR TITLE
Implement --one-based:

### PR DESCRIPTION
**Pull request by xsawyerx**
[This is my second attempt at it. I've closed #3162.]

If you now call the code with --one-based, your "jobs id" will start
from one instead of zero.

    # This sets the 2nd output to "output_file"
    split_scp.pl -j 2 1 inscp_file output_file

    # This sets the 1st output to "output_file"
    split_scp.pl --one-based -j 2 1 inscp_file output_file

I apologize for the time this super-small patch took. I found the code
confusing and had to sit and write tests to make sure I understand how
the code works and what this patch will do. Then work got in the way. :/

I have separated this completely from any other change I want to do
of clean ups. Once this is merged, I would like to start introducing
small changes that make the code more readable and more secure.

(Specifically, there are several security issues with this code that
my original PR fixed, but I will provide them separately in a clean
PR.)

Thank you for the patience and help in understanding this.

### CODE REVIEW

**Comment by danpovey** [comment]
Thanks!  @kkm000 do you have a chance to take a quick look?

**Comment by xsawyerx** [comment]
I can also provide the tests I've written in a separate file if you're interested. They run through various command line inputs and see that the variables are set correctly and then output TAP results (a series of "ok" and "not ok" with the test names) which can either be run as is or within a testing harness (`prove`) that comes with Perl.

File: egs/wsj/s5/utils/split_scp.pl [hunk 1]
```diff
@@ -44,6 +44,7 @@
 $num_jobs = 0;
 $job_id = 0;
 $utt2spk_file = "";
+$one_based = 0;
```
Comment (danpovey) [inline egs/wsj/s5/utils/split_scp.pl]: would you mind increasing the loop over x to $x <= 3?  (max num-args).  After that I think it's good to merge.  Thanks!

Comment (xsawyerx) [reply egs/wsj/s5/utils/split_scp.pl]: Of course. Done!

**Comment by danpovey** [comment]
Thanks!  Merging.

**Comment by Qoboty** [comment]
------------------ 原始邮件 ------------------
发件人: "Daniel Povey"<notifications@github.com>;
发送时间: 2019年5月2日(星期四) 凌晨2:05
收件人: "kaldi-asr/kaldi"<kaldi@noreply.github.com>;
抄送: "Subscribed"<subscribed@noreply.github.com>; 
主题: Re: [kaldi-asr/kaldi] Implement --one-based: (#3279)




Thanks!  Merging.

**Pull request merged**